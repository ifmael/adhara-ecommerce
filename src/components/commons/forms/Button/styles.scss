@import "styles/_variables.scss";

.custom-button {
  min-height: 3rem;
  color: $white;
  font-size: 1.6rem;
  text-align: center;
  background-color: $mainColor;
  padding: 0 2rem 0.1rem;
  border: 0.1rem solid $mainColor;
  border-radius: 2.5rem;
  opacity: 1;
  cursor: pointer;
  transition: opacity 300ms linear;
}
