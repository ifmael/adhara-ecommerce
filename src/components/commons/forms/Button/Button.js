import React from "react";
import { bool, func, string } from "prop-types";
import "./styles.scss";

const propTypes = {
  disabled: bool,
  onClickHandler: func,
  text: string,
};

const defaultProps = {
  disabled: false,
  onClickHandler: () => {},
  text: "",
};

const Button = ({ text, onClickHandler, disabled, children }) => {
  return (
    <button
      onClick={onClickHandler}
      className="custom-button"
      disabled={disabled}
    >
      {text}
      {children}
    </button>
  );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
