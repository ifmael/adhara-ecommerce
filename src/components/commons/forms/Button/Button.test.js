import React from "react";
import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Button test", () => {
  test("take a snapshot", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment(<Button />)).toMatchSnapshot();
  });

  test("renders button", () => {
    render(<Button text="Button" />);

    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent("Button");
  });
});
