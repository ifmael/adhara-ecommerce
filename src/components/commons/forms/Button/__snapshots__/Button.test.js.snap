// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
  />
</DocumentFragment>
`;
